segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceBuilder.java

```diff
@@ -133,9 +135,11 @@ private ServiceBuilder(ServiceBuilderConfig serviceBuilderConfig, ServiceConfig
 
         // Setup Thread Pools.
         String instancePrefix = getInstanceIdPrefix(serviceConfig);
-        this.coreExecutor = executorBuilder.apply(serviceConfig.getCoreThreadPoolSize(), instancePrefix + "core", Thread.NORM_PRIORITY);
-        this.storageExecutor = executorBuilder.apply(serviceConfig.getStorageThreadPoolSize(), instancePrefix + "storage-io", Thread.NORM_PRIORITY);
-        this.lowPriorityExecutor = executorBuilder.apply(serviceConfig.getLowPriorityThreadPoolSize(),
+        ThreadPoolExecutorConfig config = new ThreadPoolExecutorConfig(serviceConfig.getCoreThreadPoolSize(), serviceConfig.getCoreThreadPoolSize());
+        this.coreExecutor = executorBuilder.apply(config, instancePrefix + "core", Thread.NORM_PRIORITY);
+        config = new ThreadPoolExecutorConfig(serviceConfig.getStorageThreadPoolSize(), serviceConfig.getStorageMaxThreadPoolSize(), serviceConfig.getKeepAliveTime(), TimeUnit.MILLISECONDS);
```

Comment:
Either use a new variable or better yet use builder pattern in the line 141 .. something like.

```
this.storageExecutor = executorBuilder.apply(ThreadPoolExecutorConfig.builder().
                .setCoreThreadPoolSize(..)
                ...
                build(), 
                instancePrefix + "storage-io", 
                Thread.NORM_PRIORITY);
```